falcon/util/__init__.py

```diff
@@ -76,24 +76,18 @@
     (_CyBufferedReader or _PyBufferedReader) if IS_64_BITS else _PyBufferedReader
 )
 
-if PYTHON_VERSION >= (3, 7):
-    # NOTE(caselit): __getattr__ support for modules was added only in py 3.7.
-    # Deprecating an import on previous version is hard to do, so we are
-    # displaying the warning only on 3.7+
-    def __getattr__(name):
-        if name == 'json':
-            import warnings
-            import json  # NOQA
-
-            warnings.warn(
-                'Importing json from "falcon.util" is deprecated.', DeprecatedWarning
-            )
-            return json
-        from types import ModuleType
-
-        # fallback to the default implementation
-        mod = sys.modules[__name__]
-        return ModuleType.__getattr__(mod, name)
-
-else:
-    import json  # NOQA
+
+def __getattr__(name):
+    if name == 'json':
+        import warnings
+        import json  # NOQA
+
+        warnings.warn(
```

Comment:
Maybe we could just remove this?

Reply:
Good idea, but we need to add a breaking change newsfragment then telling what was removed. I can do this in a follow-up PR.

Reply:
Fine by me